pkg/distributor/distributor.go

```diff
@@ -316,6 +322,25 @@ func (d *Distributor) PushParsed(ctx context.Context, req *distributormodel.Push
 		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("no profiles received"))
 	}
 
+	for _, series := range req.Series {
+		for _, sample := range series.Samples {
+			mr := exporter.WriteRequest{
+				Timestamp: time.Unix(0, sample.Profile.TimeNanos).UnixMilli(),
+			}
+			pprof.ExtractMetrics(sample.Profile.Profile, func(labels labels.Labels, v int64) {
+				mr.TimeSeries = append(mr.TimeSeries, exporter.TimeSeries{Labels: labels, Value: float64(v)})
+			})
+
+			if len(mr.TimeSeries) > 0 {
+				go func() {
+					if sendErr := d.exporter.Send(context.Background(), &mr); sendErr != nil {
+						_ = d.logger.Log("msg", "failed to push metrics", "err", sendErr)
+					}
+				}()
+			}
+		}
+	}
+
```

Comment:
For PoC purposes only. Prometheus remote write protocol requires strict order by time within series, therefore we'll likely need a dedicated component that will handle proxying to receiver